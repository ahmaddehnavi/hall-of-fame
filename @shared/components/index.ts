export * from './layout/Col'
export * from './layout/Row'
export * from './screen/Screen'
export * from './backhandler/MultiBackHandler'
export * from './touchable/Touchable'
export * from './button/Button'
export * from './list/FlatList'
export * from './empty/EmptyView'
export * from './loading/LoadingView'
export * from './error/ErrorView'
export * from './async-view/AsyncView'