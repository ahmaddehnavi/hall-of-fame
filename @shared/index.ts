export * from './services'
export * from './utils'
export * from './components'
export * from './modules'
export * from './stores'
