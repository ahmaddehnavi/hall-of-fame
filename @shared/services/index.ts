export * from './base'
export * from './navigation'
export * from './theme'

