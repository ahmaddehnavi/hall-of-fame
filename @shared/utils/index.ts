export * from './di/DIUtil'
export * from './async/AsyncUtil'