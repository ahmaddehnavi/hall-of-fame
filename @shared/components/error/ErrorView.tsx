import {Button, Col, ListResource, Resource,} from '@shared';
import {observer} from 'mobx-react';
import * as React from 'react';
import {StyleProp, StyleSheet, Text, ViewStyle} from 'react-native';
import {BaseResource} from '../../modules/resource/base/BaseResource';

type ADFlatListRetryProps = {
    resource?: BaseResource<any, any> | Resource<any, any> | ListResource<any, any>,
    /**
     * if not set use resource#reload
     */
    onRetryPress?: () => void
    /**
     * if not set use resource#message
     */
    message?: string,
    style?: StyleProp<ViewStyle>
}

@observer
export class ErrorView extends React.Component<ADFlatListRetryProps> {

    get message() {

        if (this.props.message) {
            return this.props.message
        }

        let res = this.props.resource;
        if (res) {
            if (res instanceof ListResource || res instanceof Resource) {
                if (res.isError) {
                    return res.error.message || String(res.error);
                }
                return res.message
            }
        }

        return undefined
    }

    render() {
        let {resource} = this.props;
        if (resource && !resource.isError) {
            return false
        }

        return (
            <Col style={[styles.container, this.props.style]}>
                <Text style={{textAlign: 'center', margin: 16}}>
                    {this.message}
                </Text>
                {
                    !!(this.props.onRetryPress || resource) && (
                        <Button
                            filled
                            accent
                            title={'retry'}
                            onPress={this.props.onRetryPress || (resource ? resource.reload : undefined)}/>
                    )
                }
            </Col>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'center',
        flexGrow: 1,
    }
});