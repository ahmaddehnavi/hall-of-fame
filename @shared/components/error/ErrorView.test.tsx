import {NavigationService} from '@shared';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {ErrorView} from './ErrorView';

describe('ErrorView Tests', () => {

    it('should match snapshot', () => {
        let tree = renderer.create(
            <ErrorView

            />
        );
        expect(tree).toMatchSnapshot();
    });

});
