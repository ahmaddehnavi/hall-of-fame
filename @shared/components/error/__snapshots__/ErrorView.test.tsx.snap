// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView Tests should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "margin": 16,
        "textAlign": "center",
      }
    }
  />
</View>
`;
